pkg/operator/controller/dns/controller.go

```diff
@@ -735,14 +749,29 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 		}
 	case configv1.PowerVSPlatformType:
 		// Power VS platform will use the ibm dns implementation
+
+		// Read custom service endpoints from the infra status and set when configured
+		// When none configured, the provider will use the default values
+		var serviceEndpointOverrides ibm.ServiceEndpointOverrides
+		for _, endpointConfig := range infraStatus.PlatformStatus.PowerVS.ServiceEndpoints {
+			switch endpointConfig.Name {
+			case "cis":
+				serviceEndpointOverrides.CIS = endpointConfig.URL
+			case "dnsservices":
+				serviceEndpointOverrides.DNS = endpointConfig.URL
+			case "iam":
+				serviceEndpointOverrides.IAM = endpointConfig.URL
+			}
+		}
```

Comment:
I was going to suggest writing a helper using Go generics, but it turns out that generics isn't flexible enough for this case (see golang/go#48522).  Given that, I'll just note that the comment here should be updated per my earlier suggestion.